src/plugins/media/media.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * Media Plugin
+ * 
+ * This plugin will do the following things:
+ * 
+ *  - Add a special class when playing and pausing media (removing them when ending).
```

Comment:
More detail: Add a special class `body.impress-media-playing` and `body.impress-media-paused`.

Also document what classes exists on body when presentation has multiple media and some are playing and others are paused.

Reply:
I would only add classes for media on the current step.
Multiple media of the same step being playing or paused at the same time is indeed an issue. Potentially causing head ache for users …

`body.impress-media-playing`, `body.impress-media-paused`, `body.impress-media-id-[ID]-playing` and `body.impress-media-id-[ID]-paused` could be a solution to target specific media. But the body tag is already quite cluttered with class names, so is it really necessary, to add some more?

Reply:
That's a good question :-) I'm just happy to see the autoplay and autostop!

I guess one question is, do you really need classes for paused/stopped? What if there was just a single class `body.impress-media-playing` which is set when at least one media element - perhaps restricted to current step - is playing?

I could also see usefulness in separating audio and video: `impress-media-video-playing` and `impress-media-audio-playing`

Reply:
I like a `*paused` class because I darken the background or fade out other elements when playing videos, but I don’t want them to fade in, when I pause a video for a moment to give some extra explanation. This would not be possible, if there is no class for that. Of course it is up to the user to revert to original styling, when one does not make use of that possibility.

Video and audio are already separated, because maybe one does not want to darken the background when playing audio, but only when playing video (seems logical to me).

Reply:
But if you have both `*playing* and `*paused` classes and multiple videos or audios at one step it can have the strange effect, that both classes are present at the same time.
Of course one could remove that other class first (this is the current implementation), but this is maybe not very intuitive behaviour, because in fact you have both states at the same time.

Reply:
So if you have: `impress-media-video-playing, impress-media-video-paused` (and same for audio), but no `-stopped` class, then you end up with:

* -playing = at least one video (or audio) is playing
* -paused = at least one video (audio) is paused
* Nothing = no video (audio) exists, or if it does, it is in stopped state

Those would be clear to me. Note that authoring meaningful CSS for the case where multiple videos are playing or paused simultaneously may not be easy or possible. IMO that's ok. An author can also write their own javascript to catch events and do whatever they want, if it's no possible from above menu.